### ISSUE TITLE
MD007: incorrectly warns when mixing ul/ol

### ISSUE THREAD

**Issue by EvaSDK** [jackdewinter/pymarkdown#975]
I have a few documents that should be valid but are reporting issues in pymarkdown due to what appears to be a problem related to mixing types of lists.

Here is the sample file:

```markdown
# Lorem Ipsum

- Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    - Sed sit amet tellus interdum, fringilla sem id, porta dolor.
    - Pellentesque posuere felis vel est scelerisque, eget ultrices enim pellentesque.

1. Phasellus porta leo quis libero feugiat, vel interdum massa lobortis.
    - Nam eget metus nec ante lacinia accumsan id eu libero.
    - Nulla tristique tortor et suscipit condimentum.
```

Here is configuration file:
```json
{
  "plugins": {
    "ul-indent": {
      "indent": 4
    }
  }
}
```

Running pymarkdown returns:
```console
$ pymarkdown version
0.9.16
$ pymarkdown scan test.md
test.md:8:5: MD007: Unordered list indentation [Expected: 4, Actual=5] (ul-indent)
test.md:9:5: MD007: Unordered list indentation [Expected: 4, Actual=5] (ul-indent)
```

This issue is not new (had the same with 0.9.13.4). markdownlint has no issue with this document but I try to avoid it as I'm using this tool in pre-commit and I don't want to bother getting NPM in my CI for this.

**Comment by EvaSDK** [comment]
Forgot to mention the reason I'm using 4-spaces indent is because the documents are meant to be consumed by mkdocs so it goes through mdformat-mkdocs which enforces this.

**Comment by horejsek** [comment]
I experience the same issue.

**Comment by jackdewinter** [comment]
Just noticed that we had that issue a while back https://github.com/jackdewinter/pymarkdown/issues/301... looking into it.

**Comment by jackdewinter** [comment]
Actually, downloaded the tool and tried something out.  Can I get you to verify what happens when you copy this snippet to a file and scan it through the tool?

```Markdown
- abc
  - def

1. abc
   - def

> - abc
>   - def
>
> 1. abc
>    - def
```

**Comment by EvaSDK** [comment]
Here is what I get:
```
$ npm install markdownlint-cli
$ npx markdownlint issue975-issuecomment-1920378562.md
issue975-issuecomment-1920378562.md:1 MD041/first-line-heading/first-line-h1 First line in a file should be a top-level heading [Context: "- abc"]
$ pymarkdown scan issue975-issuecomment-1920378562.md
issue975-issuecomment-1920378562.md:1:1: MD041: First line in file should be a top level heading (first-line-heading,first-line-h1)
```

**Comment by jackdewinter** [comment]
@EvaSDK  Sorry, I meant `mkdocs`.  When I did it here, it made sure that both ordered and unordered lists stuck to that 4 indent basis.  

Md007 only deals with unordered lists, and changing that did not sit well with me.  So, I created a new PML101 that uses "anchored list indents" that seems to work in all the cases I have thrown at it, validated by mdformat and mkdocs.  It should be released within a couple of days.
